client/data/timeline/resolvers.ts

```diff
@@ -10,13 +10,25 @@ import { apiFetch } from '@wordpress/data-controls';
  */
 import { NAMESPACE } from '../constants';
 import { updateTimeline, updateErrorForTimeline } from './actions';
+import { Timeline } from './types';
 
-export function* getTimeline( intentionId ) {
+const isTimeline = ( responseData: any ): responseData is Timeline => {
+	return (
+		typeof responseData === 'object' &&
+		undefined !== ( responseData as Timeline ).length
+	);
```

Comment:
Same as with charges, I don't fully understand the purpose of this check

Reply:
It would be best to refer to my other reply: https://github.com/Automattic/woocommerce-payments/pull/1280#discussion_r583842155

To be fair though; I'm not sure this is the best way to do this type guard. I'd love to get some recommendations there.

Reply:
Well, the decision is the same as above. So it's fine to go as is, I'm only curious what happens to UX if this check fails. I.e. neither update nor error action are triggered. Will it stuck in loading state? If so we'd probably need to generate/throw error in else.

Reply:
As per previous discussions, [we now throw an error](https://github.com/Automattic/woocommerce-payments/blob/6d0a079b71a903357d47f69c4238258dd4bc8281/client/data/charges/resolvers.ts#L27-L29) when the type guard fails 🙂